spirv_msl.cpp

```diff
@@ -5615,6 +5615,10 @@ void CompilerMSL::emit_custom_templates()
 // otherwise they will cause problems when linked together in a single Metallib.
 void CompilerMSL::emit_custom_functions()
 {
+	// Use when outputting overloaded functions to cover different address spaces.
+	static const char *addr_spaces[] = { "device", "constant", "thread", "threadgroup" };
```

Comment:
threadgroup is non-sensical. Is that really supposed to work?

Reply:
I set up `addr_spaces` to apply generically, and figured an extra overloaded function wouldn't matter. However, your point is valid, and I can see that it might sow confusion here and there.

I've modified it to apply to `graphics_args_addr_spaces` only, and removed `threadgroup`. If a similar `compute_args_addr_spaces` list arises in the future, we can add it separately to include `threadgroup`.

Reply:
Can you confirm that main right now is okay? Need to know for SDK release.

Reply:
I just re-pulled from `main`. 

`main` has `texture_addr_spaces`, which is fine.

Reply:
And `test_shaders.sh` runs fine.